### PR TITLE
(testing): Bump K8S_VERSION for testing to 1.26.0

### PR DESCRIPTION
**Pull request by everettraven**
**Description of the change:**
- Bumps `K8S_VERSION` in Makefile to 1.26.0 so that our testing environment uses k8s 1.26
- Bumps `kind` version in Makefile to `0.17.0` so that it is up to date with the latest release
- Bumps `golangci-lint` version in Makefile to `1.51.2` to fix a problem I had locally where running `make lint` resulted in my terminal or editor crashing.

**Motivation for the change:**
- K8s 1.26 bump

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [ ] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [ ] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

**Review by rashmigottipati — APPROVED** [review]
Changes look good to me. 
However, IMO, we should wait to merge this until SDK k8s bump PR gets merged first.

**Review by rashmigottipati — COMMENTED** [review]
/hold

File: Makefile
Comment (varshaprasad96) [inline Makefile:60]: Since we are on this, there were a bunch of deprecated linters which we seem to still use.

```
WARN [runner] The linter 'structcheck' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter.  Replaced by unused.
WARN [runner] The linter 'maligned' is deprecated (since v1.38.0) due to: The repository of the linter has been archived by the owner.  Replaced by govet 'fieldalignment'.
WARN [runner] The linter 'interfacer' is deprecated (since v1.38.0) due to: The repository of the linter has been archived by the owner.
WARN [runner] The linter 'varcheck' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter.  Replaced by unused.
WARN [runner] The linter 'deadcode' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter.  Replaced by unused.
WARN [runner] The linter 'golint' is deprecated (since v1.41.0) due to: The repository of the linter has been archived by the owner.  Replaced by revive.
```
Could we also replace the available alternative ones in https://github.com/operator-framework/operator-sdk/blob/master/.golangci.yml.

Comment (everettraven) [reply Makefile:60]: Done in commit https://github.com/operator-framework/operator-sdk/pull/6351/commits/ebf16c8e09be81c419c1d2fc15529c03a43c5f56

**Review by rashmigottipati — APPROVED** [review]
/lgtm

File: .golangci.yml
Comment (varshaprasad96) [inline .golangci.yml:9]: Could we also add [unused](https://github.com/golangci/golangci-lint/blob/master/pkg/golinters/unused.go), a suggested replacement for dead code and varcheck

**Review by varshaprasad96 — APPROVED** [review]
Just one nit, else 
/lgtm

File: .golangci.yml
Comment (everettraven) [reply .golangci.yml:9]: Already exists here: https://github.com/operator-framework/operator-sdk/pull/6351/files#diff-6179837f7df53a6f05c522b6b7bb566d484d5465d9894fb04910dd08bb40dcc9R15

**Pull request merged**
